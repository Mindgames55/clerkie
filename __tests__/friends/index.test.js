import React from "react";
import { render } from '@testing-library/react';
import Friends from '@/app/friends/page';

// need to mock the server component as jest doesn't support async components
jest.mock('../../components/friends/ServerFriendsList', () => () => {
  const ReactComponent = React.forwardRef((props, ref) => <span ref={ref} {...props} />);
  ReactComponent.displayName = 'ReactComponentMock'
})

describe('Friends', () => {
  it('renders correctly', async () => {
    const { container } = render(<Friends />);
    expect(container).toMatchSnapshot();
  });
});