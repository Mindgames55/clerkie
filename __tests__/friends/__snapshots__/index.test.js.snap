// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <div
        class="container"
      >
        <div
          class="controls"
        >
          <button
            aria-label="filter by status"
            class="filter"
          >
            <span />
          </button>
          <span
            class="separator"
          >
            |
          </span>
          <button
            class="clear"
            disabled=""
          >
            Clear all
          </button>
        </div>
        <div
          class="container hidden"
        >
          <header
            class="header"
          >
            <div
              class="box"
            >
              <button
                class="clear"
                disabled=""
              >
                Clear all
              </button>
            </div>
            <div
              class="box"
            >
              <span
                class="title"
              >
                Filter
              </span>
            </div>
            <div
              class="box"
            >
              <button
                aria-label="close"
              >
                <span />
              </button>
            </div>
          </header>
          <div
            class="filter-container"
          >
            <p
              class="status-title"
            >
              Friend Status
            </p>
            <ul
              class="options"
            >
              <li
                class="option"
              >
                <label
                  class="option-copy"
                  for="close"
                >
                  Close Friends
                </label>
                <input
                  class="checkbox"
                  id="close"
                  name="close"
                  type="checkbox"
                />
              </li>
              <li
                class="option"
              >
                <label
                  class="option-copy"
                  for="super-close"
                >
                  Super Close Friends
                </label>
                <input
                  class="checkbox"
                  id="super-close"
                  name="super-close"
                  type="checkbox"
                />
              </li>
            </ul>
            <button
              class="apply"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
      <ul>
        <div>
          <span
            class="loader"
          />
        </div>
      </ul>
    </div>
  </div>
</div>
`;
